### ISSUE TITLE
StakingPool unit tests: requestAllocations

### ISSUE THREAD

**Issue by roxdanila** [NexusMutual/smart-contracts#492]
**!! The 2 functions below have been merged into one - test cases need an update**

**allocateStake**
- [ ] reverts if cover contract is not the caller
- [ ] updates tranches including accNxmPerRewardsShare and lastAccNxmUpdate
- [ ] reverts if insufficient capacity as a result of burns
- [ ] reverts if insufficient capacity for requested amount
- [ ] updates allocated capacities in activeCoverAmounts
- [ ] updates expiring cover amounts in coverTrancheAllocations and expiringCoverBuckets
- [ ] updates product price
- [ ] reverts if reward ratio exceeds denominator
- [ ] updates bucket reward per second cut
- [ ] correctly calculates premium
- [ ] correctly calculates rewards

**deallocateStake**
- [ ]  if cover contract is not the caller
- [ ] updates tranches including accNxmPerRewardsShare and lastAccNxmUpdate
- [ ] updates allocated capacities in activeCoverAmounts
- [ ] updates expiring cover amounts in coverTrancheAllocations and expiringCoverBuckets
- [ ] updates bucket reward per second cut

**Comment by roxdanila** [comment]
Blocked by https://github.com/NexusMutual/smart-contracts/issues/557
